### ISSUE TITLE
Deploy local commands: rsync permissions

### ISSUE THREAD

**Issue by swalkinshaw** [roots/trellis#303]
See https://github.com/roots/roots-example-project.com/issues/22 for more details.

/cc @fullyint

**Comment by fullyint** [comment]
@austinpray pointed out that "the [synchronize](https://docs.ansible.com/ansible/synchronize_module.html) module uses the `-a` flag by default."
"This is equivalent to **-rlptgoD**" ([man page](http://manpages.ubuntu.com/manpages/trusty/en/man1/rsync.1.html)).

**-r, --recursive**: (keep)
**-l, --links**: (keep) copy symlinks as symlinks
**-p, --perms**: (drop) this is the problem, so we'll set the parameter `perms=no`
**-t, --times**: (keep) preserve modification times
**-g, --group**: (drop) preserve group. We'll set the parameter `group=no`.
**-o, --owner**: (drop) preserve owner (super user only). We'll set the parameter `owner=no` just to be safe (?).
**-D** The -D option is equivalent to --devices --specials
**--devices**: (keep?) "transfer character and block device files to the remote system to recreate these devices"
**--specials**: (keep?) "transfer special files such as named sockets and fifos"

In my tests source permissions were synced over, but the owner:group seemed to always end up `web:www-data`. I tried adding `perms=no group=no owner=no` to [our synchronize task](https://github.com/roots/trellis/blob/f18d68fea6501b9e2180173845209b4489607a9b/roles/deploy/tasks/main.yml#L33) and the read and write permissions were no longer synced. I don't understand, however, why the executability was always synced, even if I ran the rsync command manually. Maybe it's just my machine or something I'm not understanding. 

I'm pretty inexperienced with permissions and I'm left wondering whether some people would actually want their local files' custom permissisons to sync over (?). 

If we want to explicitly set permissions, adding the following would set directories to 755 and files to 644:
`rsync_opts=--chmod=Du=rwx,--chmod=Dg=rx,--chmod=Do=rx,--chmod=Fu=rw,--chmod=Fg=r,--chmod=Fo=r`
(I didn't have to repeat the `--chmod=` on a manual rsync, but it seemed necessary with the Ansible synchronize module.)

Would we want to set permissions explicitly? Are those the permissions we'd want (D755,F644)?

**Comment by swalkinshaw** [comment]
I don't think we should ever have local permissions being transferred over by default. I'd say it's just a bad practice since you wouldn't expect it and could lead to issues.

644 should be good.

**Comment by fullyint** [comment]
Here's where I'm headed, in case anyone has feedback: 
```diff
 - name: Copy project local files
-  synchronize: src="{{ item.src }}" dest="{{ deploy_helper.new_release_path }}/{{ item.dest }}"
+  synchronize: src="{{ item.src }}"
+               dest="{{ deploy_helper.new_release_path }}/{{ item.dest }}"
+               group=no
+               owner=no
+               rsync_opts=--chmod=Du=rwx,--chmod=Dg=rx,--chmod=Do=rx,--chmod=Fu=rw,--chmod=Fg=r,--chmod=Fo=r
   with_items: project_local_files
```
* `perms=no` is unnecessary when the `--chmod=` options are included
* `group=no` should be necessary, but on my setup, owner and group never get synced, so I'm unsure.
* `owner=no` seems less critical because it applies only if super user.

At this point, I'd play it safe and include both `group=no` and `owner=no`. Feedback and testing welcome.

**Comment by swalkinshaw** [comment]
@fullyint looks good :+1:

**Comment by swalkinshaw** [comment]
Fixed by #309
